### PR TITLE
Update rust according to moz-central updates [ci full]

### LINKED ISSUE #4485: Update to rust 1.55
[mozilla-central now "uses" 1.55](https://bugzilla.mozilla.org/show_bug.cgi?id=1727470) - according to [our policy](https://github.com/mozilla/application-services/blob/main/docs/rust-versions.md) this means we should too. Sammy is working on a clippy fallout PR so maybe we can bump at the same time?

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-2794)
┆Sprint End Date: 2021-09-30

### PR DESCRIPTION
**Pull request by skhamis**
fixes #4485,

According to our [Rust policy](https://github.com/mozilla/application-services/blob/main/docs/rust-versions.md#application-services-rust-version-policy) we should be updating our toolchain to keep up with moz-central.  

Note: Most of the changes came from running `cargo clippy --fix` and some hand-changed but almost all of them are dealing the error:

```
error: this expression borrows a reference that is immediately dereferenced by the compiler
```

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ commands:
   setup-rust-min-version:
     steps:
       # Technically should be on 1.47 here, but that boat sailed - 1.50 has been our effective min for some time now.
-      - run: rustup override set 1.50.0
+      - run: rustup override set 1.53.0
```
Comment (mhammond) [inline .circleci/config.yml:16]: See https://searchfox.org/mozilla-central/source/python/mozboot/mozboot/util.py#20 - this should stay at 1.53.0 (so the only change here should be the removal of the comment)

**Review by mhammond — CHANGES_REQUESTED** [review]

**Comment by mhammond** [comment]
looks good to me in general, but flagging some nimbus folk

File: .circleci/config.yml (hunk 1 above)
Comment (mhammond) [reply .circleci/config.yml:16]: (oops - sorry - this comment is confused and wrong! The change is correct, but please remove the comment)

Comment (skhamis) [reply .circleci/config.yml:16]: hah figured what you meant! Removed the comment, I did add the you posted above though, helpful context

File: rust-toolchain [hunk 2]
```diff
@@ -7,14 +7,13 @@
 #   supported and used in CI.
 
 [toolchain]
-channel = "1.52.1"
+channel = "1.55.0"
```
Comment (skhamis) [inline rust-toolchain:10]: Given that there is quite a few (mechanical) changes. @jhugman, @travis79  -- this is where the main rust version bump actually comes from

**Review by travis79 — COMMENTED** [review]
That's a lot of borrows being returned! I don't have any objection or concerns that can see, carry on!

**Review by mhammond — APPROVED** [review]

**Pull request merged**